mvrx/src/main/kotlin/com/airbnb/mvrx/PersistState.kt

```diff
@@ -124,6 +148,25 @@ internal fun <T : Any> Bundle.restorePersistedState(initialState: T): T {
  */
 @VisibleForTesting
 object PersistStateTestHelpers {
-    fun <T : Any> persistState(state: T) = state.persistState(assertCollectionPersistability = true)
-    fun <T : Any> restorePersistedState(bundle: Bundle, initialState: T) = bundle.restorePersistedState(initialState)
-}
\ No newline at end of file
+    fun <T : MvRxState> persistState(state: T) = state.persistState(validation = true)
+    fun <T : MvRxState> restorePersistedState(bundle: Bundle, initialState: T, validation: Boolean = false) = bundle.restorePersistedState(initialState, validation)
+}
+
+/**
+ * Throws [NoSuchElementException] if there is no method.
+ */
+@Suppress("UNCHECKED_CAST")
+private fun <T : Any> Class<T>.copyMethod(): Method = this.declaredMethods.first { it.name == "copy\$default" }
+
+private val Class<*>.defaultParameterValue: Any?
```

Comment:
Do you need to handle boxed vs not boxed here? We had to handle that in Moshi because it would fail at runtime otherwise.

https://github.com/square/moshi/blob/master/kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/kotlintypes.kt#L84-L91